<launch>
  <!-- Common parameters -->
  <arg name="gui" default="false" />
  <arg name="camera_name" default="camera" />
  <arg name="frame_rate" default="170" />
  <arg name="exposure"      value="0.004" />
  <arg name="auto_exposure"      value="false" />
  <arg name="auto_gain"     value="false" />
  <arg name="gain"          value="18" />
  <arg name="enable_trigger"          value="true" />
  <arg name="trigger_source"          value="GPIO0" />
  <arg name="trigger_polarity"          value="0" />

  <arg name="camera_serial_l" default="16276501" />
  <arg name="camera_calibrated_l" default="true" />

  <arg name="camera_serial_r" default="16276527" />
  <arg name="camera_calibrated_r" default="true" />

  <group ns="$(arg camera_name)" >
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />

    <!-- Both cameras are not loaded into one nodelet manager to avoid the namespacing issue. -->
    <!-- node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyStereoCameraSPNodelet camera_nodelet_manager" launch-prefix="valgrind -v" -->
    <node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyStereoCameraSPNodelet camera_nodelet_manager">
      <param name="frame_id_l" value="camera_l" />
      <param name="frame_id_r" value="camera_r" />
      <param name="serial_l" value="$(arg camera_serial_l)" />
      <param name="serial_r" value="$(arg camera_serial_r)" />
      <param name="camera_id_l" value="0" />
      <param name="camera_id_r" value="1" />

      <!-- When unspecified, the driver will use the default framerate as given by the
           camera itself. Use this parameter to override that value for cameras capable of
           other framerates. -->
      <param name="frame_rate" value="$(arg frame_rate)" />
      <param name="auto_exposure" value="$(arg auto_exposure)" />
      <param name="exposure" value="$(arg exposure)" />
      <param name="auto_gain" value="$(arg auto_gain)" />
      <param name="gain" value="$(arg gain)" />
      <param name="enable_trigger" value="$(arg enable_trigger)" />
      <param name="trigger_source" value="$(arg trigger_source)" />
      <param name="trigger_polarity" value="$(arg trigger_polarity)" />

      <!-- Use the camera_calibration package to create this file -->
      <param name="camera_info_url_l" if="$(arg camera_calibrated_l)"
             value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial_l).yaml" />
      <param name="camera_info_url_r" if="$(arg camera_calibrated_r)"
             value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial_r).yaml" />
    </node>

    <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
          args="load image_proc/debayer camera_nodelet_manager">
    </node>

    <include file="$(find stereo_image_proc)/launch/stereo_image_proc.launch">
        <arg name="manager" value="/camera_nodelet_manager" />
    </include>

  </group>
  <!-- group if="$(arg gui)" >
    <node name="left_image" pkg="image_view2" type="image_view2" >
      <param name="image_transport" value="compressed" />
      <remap from="image" to="/camera/left/image_rect_color"  />
    </node>
    <node name="right_image" pkg="image_view2" type="image_view2" >
      <param name="image_transport" value="compressed" />
      <remap from="image" to="/camera/right/image_rect_color" />
    </node>
  </group -->
</launch>
